apps/meteor/app/e2e/client/rocketchat.e2e.js

```diff
@@ -402,6 +411,64 @@ class E2E extends Emitter {
 	closeAlert() {
 		banners.closeById('e2e');
 	}
+
+	async getQuoteMessages(message) {
+		const { msg } = message;
+		const urls = parseUrlsInMessage(msg);
+
+		urls.map(async (url) => {
+			if (!url.includes(Meteor.absoluteUrl())) {
+				return;
+			}
+
+			const urlObj = URL.parse(url);
+			// if the URL doesn't have query params (doesn't reference message) skip
+			if (!urlObj.query) {
+				return;
+			}
+
+			const { msg: msgId } = QueryString.parse(urlObj.query);
+
+			if (!msgId) {
+				return;
+			}
+
+			const getQuotedMessage = await APIClient.get('/v1/chat.getMessage', { msgId });
+			const quotedMessage = getQuotedMessage?.message;
+
+			if (!getQuotedMessage?.success || !quotedMessage) {
+				return;
+			}
+
+			const decryptedQuoteMessage = await this.decryptMessage(quotedMessage);
+
+			message.attachments = message.attachments || [];
+
+			const quoteAttachment = {
```

Comment:
I think we do something similar in backend for regular messages, right? Is there any possibility we could implement a function to generate an attachment from a message and use it in both places? We could event unit test this function after...